UnitTests/MemoryStreamUnitTests/Write.cs

```diff
@@ -385,6 +385,28 @@ public void BoundaryCheck()
             }
         }
 
+        [TestMethod]
+        public void Write_AtMaxLength_Works()
+        {
+            var maxLength = 0xFFFF;
+            using (var ms = new MemoryStream())
+            {
+                ms.Position = maxLength - 1;
+                ms.WriteByte(0); // should resize the buffer to the max size
```

Comment:
by convention, in nano, we try to place the comment before the code

Reply:
Damn, I already did that in one of the last PRs, one day I'll get used to it.